azavea/loam
# Use GitHub actions for CI

## Overview

Adds GitHub actions for building PRs, tags, and pushes to master and develop.

### Notes

- There's a fair bit going on here. I've used templates from the Actions documentation, which are fairly well commented, but if you have questions, please ask!

## Testing Instructions

 * Open a test PR with a dummy commit, make sure it builds.
 * Alter one of the tests so that it fails (the easiest way is to change the expectation conditions). Confirm that the build fails after pushing that commit.
 * Create a tag that starts with a number, e.g. `9.test-actions`, and push it.
 * Confirm that a `Release` action runs, and that the log output indicates that NPM and GitHub will receive the same artifacts. Confirm that a draft release shows up under the project releases with the proper attachments.
 * Delete your test PR, test branch, test tag, and draft commit.


## Checklist


Resolves #17
